src/main/java/com/github/tomakehurst/wiremock/matching/MatchesJsonPathPattern.java

```diff
@@ -110,6 +112,8 @@ protected MatchResult isAdvancedMatch(String value) {
     } catch (SubExpressionException e) {
       notifier().info(e.getMessage());
       return MatchResult.noMatch(SubEvent.warning(e.getMessage()));
+    } catch (NullPointerException e) {
```

Comment:
I'm not keen on catching NPE. Something like this might be better (around line 94):
```java
if (expressionResult == null || expressionResult.isEmpty()) {
        if (AbsentPattern.class.isAssignableFrom(valuePattern.getClass())) {
          expressionResult = ListOrSingle.of((String) null);
        } else {
          return MatchResult.noMatch();
        }
      }
```